cognite/client/_api_client.py

```diff
@@ -263,57 +263,115 @@ def _list_generator(
         filter: Dict = None,
         sort: List[str] = None,
         other_params: Dict = None,
+        partitions: int = None,
         headers: Dict = None,
     ):
         if limit == -1 or limit == float("inf"):
             limit = None
         cls = cls or self._LIST_CLASS
         resource_path = resource_path or self._RESOURCE_PATH
-        total_items_retrieved = 0
-        current_limit = self._LIST_LIMIT
-        if chunk_size and chunk_size <= self._LIST_LIMIT:
-            current_limit = chunk_size
-        next_cursor = None
-        filter = filter or {}
-        current_items = []
-        while True:
-            if limit:
-                num_of_remaining_items = limit - total_items_retrieved
-                if num_of_remaining_items < self._LIST_LIMIT:
-                    current_limit = num_of_remaining_items
-
-            if method == "GET":
-                params = filter.copy()
-                params["limit"] = current_limit
-                params["cursor"] = next_cursor
-                if sort is not None:
-                    params["sort"] = sort
-                res = self._get(url_path=resource_path, params=params, headers=headers)
-            elif method == "POST":
-                body = {"filter": filter, "limit": current_limit, "cursor": next_cursor, **(other_params or {})}
-                if sort is not None:
-                    body["sort"] = sort
-                res = self._post(url_path=resource_path + "/list", json=body, headers=headers)
-            else:
-                raise ValueError("_list_generator parameter `method` must be GET or POST, not {}".format(method))
-            last_received_items = res.json()["items"]
-            total_items_retrieved += len(last_received_items)
 
-            if not chunk_size:
-                for item in last_received_items:
-                    yield cls._RESOURCE._load(item, cognite_client=self._cognite_client)
-            else:
-                current_items.extend(last_received_items)
-                if len(current_items) >= chunk_size:
-                    items_to_yield = current_items[:chunk_size]
-                    current_items = current_items[chunk_size:]
-                    yield cls._load(items_to_yield, cognite_client=self._cognite_client)
+        if partitions:
+            if limit is not None:
+                raise ValueError("When using partitions, limit should be `None`, `-1` or `inf`.")
```

Comment:
This check won't catch `-1` or `inf`, unless it's preprocessed elsewhere?

```
>>> -1 is None
False
>>> import math
>>> math.inf is None
False
```

Reply:
I assume it's called from `_list_generator`, which has 
``` python
if limit == -1 or limit == float("inf"):
    limit = None
```

Reply:
Ah, didn't catch that, great 👍